src/Altinn.App.Api/Controllers/InstancesController.cs

```diff
@@ -484,10 +475,10 @@ public class InstancesController : ControllerBase
             return Created(url, instance);
         }
 
-        private async Task CopyDataFromSourceInstance(Application application, Instance targetInstance, Instance sourceInstance)
+        private async Task CopyDataFromSourceInstance(ApplicationMetadata application, Instance targetInstance, Instance sourceInstance)
```

Comment:
Agreed, I belive this is a weakness in the Application model, we have and Id containing org and appname (org/appname) but also a separate field for org. These could potentially become out of sync and return different orgs.
I have a suggestion for adding AppIdentifier to ApplicationMetadata class instead of string App